pkg/operator/ceph/cluster/cephstatus_test.go

```diff
@@ -197,3 +203,155 @@ func Test_cephStatusChecker_conditionMessageReason(t *testing.T) {
+	// Set the node to NotReady state
+	nodes, err := context.Clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	assert.NoError(t, err)
+	for _, node := range nodes.Items {
+		node.Status.Conditions[0].Status = v1.ConditionFalse
+		localnode := node
+		_, err := context.Clientset.CoreV1().Nodes().Update(ctx, &localnode, metav1.UpdateOptions{})
+		assert.NoError(t, err)
+	}
+
+	// There should be no error
+	err = c.forceDeleteStuckRookPodsOnNotReadyNodes()
+	assert.NoError(t, err)
+
+	// The pod should still exist since its not deleted.
+	p, err := context.Clientset.CoreV1().Pods(clusterInfo.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.NotNil(t, p)
+
+	// Add a deletion timestamp to the pod
+	pod.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+	_, err = clientset.CoreV1().Pods(clusterName.Namespace).Update(ctx, &pod, metav1.UpdateOptions{})
+	assert.NoError(t, err)
+
+	// There should be no error as the pod is deleted
+	err = c.forceDeleteStuckRookPodsOnNotReadyNodes()
+	assert.NoError(t, err)
+
+	// The pod should be deleted since the pod is marked as deleted and the node is in NotReady state
+	_, err = clientset.CoreV1().Pods(clusterName.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	assert.Error(t, err)
+	assert.True(t, kerrors.IsNotFound(err))
+
+	// The pod should not be deleted as it does not have the matching labels
+	_, err = clientset.CoreV1().Pods(clusterName.Namespace).Get(ctx, notDeletePod.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+}
+
+func TestGetRookPodsOnNode(t *testing.T) {
+	ctx := context.TODO()
+	clientset := optest.New(t, 1)
+	clusterInfo := client.NewClusterInfo("test", "test")
+	clusterName := clusterInfo.NamespacedName()
+	context := &clusterd.Context{
+		Clientset: clientset,
+	}
+
+	c := newCephStatusChecker(context, clusterInfo, &cephv1.ClusterSpec{})
+	labels := []map[string]string{
+		{"rook_cluster": clusterName.Name},
+		{"app": "csi-rbdplugin-provisioner"},
+		{"app": "csi-rbdplugin"},
+		{"app": "csi-cephfsplugin-provisioner"},
+		{"app": "csi-cephfsplugin"},
+		{"app": "rook-ceph-operator"},
+		{"rook-cluster": "test", "app": "csi-cephfsplugin"},
+	}
+
+	pod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod-with-no-label",
+			Namespace: clusterName.Namespace,
+		},
+	}
+	pod.Spec.NodeName = "node0"
+	_, err := context.Clientset.CoreV1().Pods(clusterName.Namespace).Create(ctx, &pod, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	for i, label := range labels {
+		pod.ObjectMeta.Name = fmt.Sprintf("pod-%d", i)
+		pod.ObjectMeta.Namespace = clusterName.Namespace
+		pod.ObjectMeta.Labels = label
+		_, err := context.Clientset.CoreV1().Pods(clusterName.Namespace).Create(ctx, &pod, metav1.CreateOptions{})
+		assert.NoError(t, err)
+	}
+
+	pods, err := c.getRookPodsOnNode("node0")
+	assert.NoError(t, err)
+	// A pod is having two matching labels and its returned only once
+	assert.Equal(t, 7, len(pods))
+
+	for _, pod := range pods {
```

Comment:
This is an okay test, but it is somewhat circular in testing part of the test setup itself. It is a better test to check for the pods by name here since we can know what pods to expect in the list. This also helps verify that a pod doesn't show in the list multiple times.

You can possibly make this a little easier and shorter by doing something like below.
```go
podNames := []string{}
for _, pod := range pods {
  podNames = append(podNames, pod.Name)
}
podNames = sort(podNames)
assert.Equal(t, []string{"pod-0", "pod-1", ..., "pod-6"}, podNames) 
```

You could build a list of expected pod names in the `for i, label := range labels {}` loop above also. Just make sure to sort them both so that they can be tested for `Equal` without issues that might result from ordering being off, which we don't care about.

Reply:
well, the list is already sorted, do we still need to sort?

Reply:
Not really. It wouldn't hurt but also isn't necessary.

Reply:
keeping the sort